### PR TITLE
[feature](spark-load)Spark load supports string type data import

### LINKED ISSUE #11926: [Bug] Spark load does not support string type data import
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

All version

### What's Wrong?

origin_img_v2_18a8eb6c-7f84-4ae3-b36c-8476347ae18g

### What You Expected?

none

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hf200012**
```
CREATE EXTERNAL RESOURCE "spark_resource"
PROPERTIES
(
"type" = "spark",
"spark.master" = "yarn",
"spark.submit.deployMode" = "cluster",
"spark.executor.memory" = "1g",
"spark.yarn.queue" = "default",
"spark.hadoop.yarn.resourcemanager.address" = "localhost:50056",
"spark.hadoop.fs.defaultFS" = "hdfs://localhost:9000",
"working_dir" = "hdfs://localhost:9000/tmp/doris",
"broker" = "my_hdfs_broker"
); 



create table test_partition_01(
id int,
name varchar(200),
age int
)
UNIQUE KEY(`id`)
DISTRIBUTED BY HASH(`id`) BUCKETS 1
PROPERTIES (
"replication_allocation" = "tag.location.default: 1"
);



LOAD LABEL demo.hive_EXTERNAL_table_load_00009
(
DATA INFILE("hdfs://VM-32-5-centos:9000/user/hive/warehouse/demo.db/test/*/*")
INTO TABLE test_partition
COLUMNS TERMINATED BY ","
FORMAT AS "csv"
)
WITH RESOURCE 'spark_resource'
(
"spark.executor.memory" = "1g",
"spark.shuffle.compress" = "true"
)
PROPERTIES
(
"timeout" = "3600"
);
```
image

# Proposed changes

Issue Number: close #11926

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java [hunk 1]
```diff
@@ -208,6 +211,25 @@ public boolean parse(String value) {
...
+    private EtlJobConfig.EtlColumn etlColumn;
+
+    public StringTypeParser(EtlJobConfig.EtlColumn etlColumn) {
+        this.etlColumn = etlColumn;
+    }
+
+    @Override
+    public boolean parse(String value) {
+        try {
+            return value.getBytes("UTF-8").length <= 1048576;
```
Comment (morningman) [inline fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java:225]: Better define this number somewhere.

File: fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java [hunk 2]
```diff
@@ -405,6 +405,18 @@ public boolean validateData(Object srcValue, EtlJobConfig.EtlColumn etlColumn, C
                     return false;
                 }
                 break;
+            case "STRING":
+            case "TEXT":
+                // TODO(zjf) padding string type
+                int strDataSize = 0;
+                if (srcValue != null && (strDataSize = srcValue.toString().getBytes(StandardCharsets.UTF_8).length)
+                        > 1048576) {
```
Comment (morningman) [inline fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java:413]: define `1048576`

Comment (hf200012) [reply fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java]: Resolved

File: fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java (hunk 1 above)
Comment (hf200012) [reply fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java]: Resolved

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**